lib/astc_encode.cpp

```diff
@@ -61,6 +61,8 @@ pthread_join(pthread_t thread, void** value) {
     return 0;
 }
 
+#else defined(_WIN32) && defined(WIN32_HAS_PTHREADS)
```

Comment:
For the sake of aesthestics please put the include pthreads.h case first. Like this

```
#if !defined(_WIN32) || defined(WIN32_HAS_PTHREADS)
#include <pthread.h>
#else
// Put implementation for Windows here.
#endif
```

For non-Windows we should explicitly include pthread.h rather than relying on it being included by some other unrelated include file.